src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,435 @@
+     *      createTtlIndexProbability: Should a TTL Index be created expressed as a probability from 0.0 to 1.0 [default: 0.001]
+     *
+     * CAUTION: The locked resource name is indexed in the _id field of the
+     * lock collection. An indexed field's value in MongoDB can be a maximum
+     * of 1024 bytes in length inclusive of structural overhead.
+     *
+     * @see https://docs.mongodb.com/manual/reference/limits/#Index-Key-Limit
+     *
+     * CAUTION: This store relies on all client and server nodes to have
+     * synchronized clocks for lock expiry to occur at the correct time.
+     * To ensure locks don't expire prematurely; the lock TTL should be set
+     * with enough extra time to account for any clock drift between nodes.
+     *
+     * If createTtlIndexProbability is set to a value greater than 0.0
+     * there will be a chance this store will attempt to create a TTL index on
+     * self::save(). If however the server version is less than MongoDB 2.2
+     * indicating TTL Indexes are unsupported, a db.lock.remove() query will
+     * be used to cleanup old locks instead. If you prefer to create your
+     * TTL Index manually you can set createTtlIndexProbability to 0.0 and
+     * optionally leverage self::createTtlIndex(int $expireAfterSeconds = 0).
+     *
+     * writeConcern, readConcern and readPreference are not specified by
+     * MongoDbStore meaning the collection's settings will take effect.
+     * @see https://docs.mongodb.com/manual/applications/replication/
+     */
+    public function __construct(Client $mongo, array $options, float $initialTtl = 300.0)
+    {
+        $this->mongo = $mongo;
+
+        $this->initialTtl = $initialTtl;
+
+        $this->options = array_merge(array(
+            'collection' => 'lock',
+            'createTtlIndexProbability' => 0.001,
+        ), $options);
+
+        if (!isset($this->options['database'])) {
+            throw new InvalidArgumentException(
+                'You must provide the "database" option for MongoDBStore'
+            );
+        }
+
+        if ($this->options['createTtlIndexProbability'] < 0.0 || $this->options['createTtlIndexProbability'] > 1.0) {
+            throw new InvalidArgumentException(sprintf(
+                '"%s" createTtlIndexProbability must be a float from 0.0 to 1.0, "%f" given.',
+                __METHOD__,
+                $this->options['createTtlIndexProbability']
+            ));
+        }
+
+        if ($this->initialTtl <= 0) {
+            throw new InvalidArgumentException(sprintf(
+                '%s() expects a strictly positive TTL. Got %d.',
+                __METHOD__,
+                $this->initialTtl
+            ));
+        }
+    }
+
+    /**
+     * Create a TTL index to automatically remove expired locks.
+     *
+     * If the createTtlIndexProbability option is set higher than 0.0
+     * (defaults to 0.001); this will be called automatically on self::save().
```

Comment:
`there will be a chance to be called...`

Reply:
updated